src/firefly/js/ui/NaifidPanel.jsx

```diff
@@ -2,100 +2,92 @@
  * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
  */
 
-import React, {PureComponent, memo} from 'react';
+import React, {memo, useRef} from 'react';
 import PropTypes from 'prop-types';
-import {get, sortBy} from 'lodash';
+import {sortBy} from 'lodash';
 import {resolveNaifidObj} from  './NaifidPanelWorker.js';
 import {SuggestBoxInputFieldView} from './SuggestBoxInputField';
 import {useFieldGroupConnector} from './FieldGroupConnector.jsx';
 import {TargetFeedback} from './TargetFeedback';
 
 
 const LABEL_DEFAULT='Moving Target Name:';
-const makeValidRet= (valid,message='') => ({valid,message});
-//const defValidator= (val) => val ? makeValidRet(true): makeValidRet(false,'Naif name not found');
-const defValidator= () => makeValidRet(true);
-const searchHistory =[];
-let naifNamevalue = '';
-
-
-class NaifidPanelView extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = {suggestions: ''};
-        this.getSuggestions = this.getSuggestions.bind(this);
-        naifNamevalue = props.value;
-    }
-
-    componentWillUnmount() {
-        if (this.removeListener) this.removeListener();
-        this.iAmMounted = false;
-    }
-
-    componentDidMount() {
-        this.iAmMounted = true;
-    }
-
-    getSuggestions(val= '') {
+
+
+function NaifidPanelView({showHelp, valid, message, examples, feedback, value, labelWidth, feedbackStyle, popStyle,
+                             label= LABEL_DEFAULT, fireValueChange, updateNaifNameValue}){
+    const searchHistory = useRef([]); // ref instead of state so that we can mutate it without causing re-renders
```

Comment:
See my previous comment about `searchHistory` we _do_ want that to be a global since it is a cache that should never empty.

Reply:
Yes I'm reverting this, it makes sense - we want search history throughout the app's lifetime, not just when this component is mounted.